akka-actor/src/main/scala/akka/event/ActorClassificationUnsubscriber.scala

```diff
@@ -15,7 +15,8 @@ import akka.util.unused
  *
  * Watches all actors which subscribe on the given event stream, and unsubscribes them from it when they are Terminated.
  */
-protected[akka] class ActorClassificationUnsubscriber(bus: ManagedActorClassification, debug: Boolean)
+// TODO DOTTY -> Something wrong is happening in the compiler
+protected[akka] class ActorClassificationUnsubscriber(bus: String, unsubscribe: ActorRef => Unit, debug: Boolean)
```

Comment:
I'm not sure how to minimize this, but the compiler go completely nuts on this detail, the pattern something like:
`file1.scala`
```scala
class A {
  val b = new B(this)
  def foo() = {}
}
```
`file2.scala`
```scala
class B(a: A) {
  def bar() = {
    a.foo()
  }
}
```

@nicolasstucki @smarter

Reply:
This is not enough details for us to say anything (what does "completely nuts" mean?).

Reply:
The compiler rejects the build saying that one of the class file is missing.
It builds though using incremental compilation by removing dependencies on the other compilation unit and re-adding them back.
I can try to to provide a branch as a repro, sorry for the lack of details.